coffee_core/src/nurse/strategy.rs

```diff
@@ -74,3 +74,39 @@ impl Handler for GitRepositoryLocallyAbsentStrategy {
         }
     }
 }
+
+/// Stategy for migration of the repository global directory
+/// to a local directory for each network, see [1]
+///
+/// This is a strategy tht return the list of network that
+/// needs to be migrated to to the new repository configuration.
+///
+/// [1] https://github.com/coffee-tools/coffee/issues/234
+pub struct CoffeeRepositoryDirCleanUp;
+
+#[async_trait]
+impl Handler for CoffeeRepositoryDirCleanUp {
+    async fn can_be_applied(
+        self: Arc<Self>,
+        coffee: &CoffeeManager,
+    ) -> Result<Option<Defect>, CoffeeError> {
+        let networks = ["testnet", "signet", "bitcoin", "liquid"];
+        // Check whether there exists a network-specific repositories folder for each network.
+        let mut directory_moving = vec![];
+        for network in networks {
+            let network_dir = format!("{}/{network}", coffee.config.root_path);
+            if !Path::exists(Path::new(&network_dir)) {
+                log::debug!("network dir `{network_dir}` not found");
+                continue;
+            }
+            let subpath_repo = format!("{}/{network}/repositories", coffee.config.root_path);
+            if !Path::exists(Path::new(&subpath_repo)) {
+                directory_moving.push((network.to_string(), subpath_repo));
```

Comment:
If we know the network, we could easily build the repositories path. 
What do you think?

Reply:
What is the purpose here to build the repositories path? I am missing something sorry

Reply:
This comment is a follow up for https://github.com/coffee-tools/coffee/pull/239#discussion_r1558110379

I mean if we know the affected networks, we can build the repositories directory path specific to this network so no need to store it 
```rust
/// (Affected network, path)
CoffeeGlobalrepoCleanup(Vec<(String, String)>),
```

Reply:
Ah, debugging porpuse for the moment, but I agree this code can be optimize